gnovm/pkg/gnomod/deprecated_dotmod.go

```diff
@@ -26,6 +26,163 @@ import (
 	"golang.org/x/mod/module"
 )
 
+type deprecatedModFile struct {
+	Draft   bool
+	Module  *modfile.Module
+	Gno     *modfile.Go
+	Replace []*modfile.Replace
+
+	Syntax *modfile.FileSyntax
+}
+
+// ParseDeprecatedDotModBytes parses and returns a gno.mod file.
+//
+// - fname is the name of the file, used in positions and errors.
+// - data is the content of the file.
+func parseDeprecatedDotModBytes(fname string, data []byte) (*deprecatedModFile, error) {
+	fs, err := modfileParse(fname, data)
+	if err != nil {
+		return nil, err
+	}
+	f := &deprecatedModFile{
+		Syntax: fs,
+	}
+	var errs modfile.ErrorList
+
+	for _, x := range fs.Stmt {
+		switch x := x.(type) {
+		case *modfile.Line:
+			f.add(&errs, nil, x, x.Token[0], x.Token[1:])
+		case *modfile.LineBlock:
+			if len(x.Token) > 1 {
+				errs = append(errs, modfile.Error{
+					Filename: fname,
+					Pos:      x.Start,
+					Err:      fmt.Errorf("unknown block type: %s", strings.Join(x.Token, " ")),
+				})
+				continue
+			}
+			switch x.Token[0] {
+			default:
+				errs = append(errs, modfile.Error{
+					Filename: fname,
+					Pos:      x.Start,
+					Err:      fmt.Errorf("unknown block type: %s", strings.Join(x.Token, " ")),
+				})
+				continue
+			case "module", "replace":
+				for _, l := range x.Line {
+					f.add(&errs, x, l, x.Token[0], l.Token)
+				}
+			}
+		case *modfile.CommentBlock:
+			if x.Start.Line == 1 {
+				f.Draft = parseDraft(x)
+			}
+		}
+	}
+
+	if len(errs) > 0 {
+		return nil, errs
+	}
+	return f, nil
+}
+
+func (f *deprecatedModFile) Migrate() (*File, error) {
+	fi := &File{}
+	if f.Module != nil {
+		fi.Module = f.Module.Mod.Path
+	}
+	if f.Gno != nil {
+		fi.Gno = f.Gno.Version
+	}
+	fi.Draft = f.Draft
+
+	// voluntarily not migrating, because not used/working the same way:
+	// - f.Replace
+	// - f.Syntax
+	return fi, nil
+}
+
+func (f *deprecatedModFile) add(errs *modfile.ErrorList, block *modfile.LineBlock, line *modfile.Line, verb string, args []string) {
+	wrapError := func(err error) {
+		*errs = append(*errs, modfile.Error{
+			Filename: f.Syntax.Name,
+			Pos:      line.Start,
+			Err:      err,
+		})
+	}
+	errorf := func(format string, args ...any) {
+		wrapError(fmt.Errorf(format, args...))
+	}
+
+	switch verb {
+	default:
+		errorf("unknown directive: %s", verb)
+
+	case "gno":
+		if f.Gno != nil {
+			errorf("repeated gno statement")
+			return
+		}
+		if len(args) != 1 {
+			errorf("gno directive expects exactly one argument")
+			return
+		} else if !reGnoVersion.MatchString(args[0]) {
+			fixed := false
+			if !fixed {
+				errorf("invalid gno version %s: must match format 1.23", args[0])
+				return
+			}
+		}
+
+		line := reflect.ValueOf(line).Interface().(*modfile.Line)
```

Comment:
what? why?

Reply:
https://github.com/moul/gno/blame/c0b0f863a387995001d4188ea2d10b086529b33b/gnovm/pkg/gnomod/parse.go#L186

The key point about this line is the prefix of the filename following my rename.